core/src/main/java/nu/yona/server/subscriptions/service/BuddyService.java

```diff
@@ -103,14 +105,21 @@ public void removeBuddy(UUID idOfRequestingUser, UUID buddyID, Optional<String>
 		User user = User.getRepository().findOne(idOfRequestingUser);
 		Buddy buddy = getEntityByID(buddyID);
 
-		clearMessagesAndSendDropBuddyMessage(user, buddy, message, DropBuddyReason.USER_REMOVED_BUDDY);
+		removeMessagesSentByBuddy(user, buddy);
+		removeBuddyInfoFromBuddy(user, buddy, message, DropBuddyReason.USER_REMOVED_BUDDY);
 
 		user.removeBuddy(buddy);
 		User.getRepository().save(user);
 	}
 
+	private void removeMessagesSentByBuddy(User user, Buddy buddy)
+	{
+		removeNamedMessagesSentByUser(user, buddy.getVPNLoginID());
+		removeAnonymousMessagesSentByUser(user.getAnonymized(), buddy.getVPNLoginID());
+	}
+
 	@Transactional
-	void clearMessagesAndSendDropBuddyMessage(User requestingUser, Buddy requestingUserBuddy, Optional<String> message,
+	void removeBuddyInfoFromBuddy(User requestingUser, Buddy requestingUserBuddy, Optional<String> message,
```

Comment:
You're not removing the buddy from the buddy but for the given buddy. Renaming this into ``removeBuddyInfoForBuddy``.